airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/commands.py

```diff
@@ -144,6 +149,11 @@ async def test(
         update_global_commit_status_check_for_tests(ctx.obj, "failure")
         return False
 
+    finally:
+        if RegressionTests.regression_tests_artifacts_dir.exists():
+            shutil.rmtree(RegressionTests.regression_tests_artifacts_dir)
+            main_logger.info(f"  Test artifacts cleaned up from {RegressionTests.regression_tests_artifacts_dir}")
```

Comment:
nit: another way of doing it which would align with a `teardown` phase would be to clean up in `ConnectorContext.__aexit__`. But it'd mean making the `regression_tests_artifacts_dir` connector specific.